CMakeLists.txt

```diff
@@ -1,7 +1,11 @@
-cmake_minimum_required (VERSION 2.6)
+cmake_minimum_required (VERSION 3.8.2)
```

Comment:
update cmake version for using C++17 std::optional

Reply:
👍